File: MYCA/myca/support/src/main/java/cdiomyc/support/domain/ddd/AggregateRoot.java
package cdiomyc.support.domain.ddd;

/**
 * Generic Interface for marking an aggregate root
 * @author <a href="https://github.com/freitzzz">freitzzz</a>
 * @param <EID> Generic-Type of the aggregate root domain entity identifier
 */
public interface AggregateRoot<EID> extends DomainEntity<EID>{}

File: MYCM/support/domain/ddd/AggregateRoot.cs
using support.domain.ddd;

namespace support.domain.ddd{
    /// <summary>
    /// Generic Interface for marking an entity which is the root of aggregated entities
    /// </summary>
    /// <typeparam name="EID">Generic-Type of the entity identifier</typeparam>
    public interface AggregateRoot<EID>:DomainEntity<EID>{}
}